app/src/main/java/emu/skyline/settings/SettingsActivity.kt

```diff
@@ -102,6 +107,57 @@ class SettingsActivity : AppCompatActivity() {
             .commit()
     }
 
+    override fun onCreateOptionsMenu(menu : Menu?) : Boolean {
+        menuInflater.inflate(R.menu.settings_menu, menu)
+        val menuItem = menu!!.findItem(R.id.app_bar_search)
+        val searchView = menuItem.actionView as SearchView
+        searchView.queryHint = getString(R.string.search)
+
+        searchView.setOnQueryTextFocusChangeListener { _, focus ->
+            (binding.titlebar.toolbar.layoutParams as AppBarLayout.LayoutParams).scrollFlags =
+                if (focus)
+                    AppBarLayout.LayoutParams.SCROLL_FLAG_NO_SCROLL
+                else
+                    AppBarLayout.LayoutParams.SCROLL_FLAG_SCROLL
+        }
+
+        searchView.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
+            override fun onQueryTextSubmit(query : String) : Boolean {
+                return false
+            }
+
+            override fun onQueryTextChange(newText : String) : Boolean {
+                val queries = newText.split(",").toTypedArray()
+                if (newText.isNotEmpty()) {
```

Comment:
There's an issue here when the user only inputs `,`, no setting will show up. Ideally when that happens, all settings should show up as the user is searching for "empty strings", which occur in every string. By handling this correctly, you also avoid the need to check if `newText` is empty. Here's how to fix:
```suggestion
                val queries = newText.split(",").toTypedArray()
```

Also you don't need a `TypedArray` here.